src/main/java/com/toptrumps/core/card/Dealer.java

```diff
@@ -12,7 +14,9 @@
     private final ArrayList<Card> deck;
 
     public Dealer(String deckFile) {
+
         this.deck = DeckParser.parseDeck(deckFile);
+        Logger.logToFile("Deck before shuffling: \n" + printDeck());
```

Comment:
I think there's an issue with the logger's design. You have a class that should be instantiated and enabled in order to activate its logging capabilities. However, someone can just call its ```static``` method ```logToFile()``` and hope it will log stuff. 

I see some ways out of it:
1 - Have the ```Logger``` be passed to the controller (like our initial UML Class diagram) and create a convenience method that verifies whether you have a logger member instance inside the controller.
2 - Apply the ```singleton``` pattern so you have only one instance of ```Logger``` across the application.
3 - Turn everything static.